Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.Model.swift

```diff
@@ -0,0 +1,140 @@
+                            attributes: decodable.attributes.makeDictionary()
+                        )
+                    }
+                
+                return VesselFeed(events: feedEvents)
+            }
+            
+            let customDataSource = CustomDynamicEntityDataSource(info: info) { try makeFeed() }
+            
+            dynamicEntityLayer = DynamicEntityLayer(dataSource: customDataSource)
+            
+            let trackDisplayProperties = dynamicEntityLayer.trackDisplayProperties
+            trackDisplayProperties.showsPreviousObservations = true
+            trackDisplayProperties.showsTrackLine = true
+            trackDisplayProperties.maximumObservations = 20
+            
+            let labelDefinition = LabelDefinition(
+                labelExpression: SimpleLabelExpression(simpleExpression: "[VesselName]"),
+                textSymbol: TextSymbol(color: .red, size: 12)
+            )
+            labelDefinition.placement = .pointAboveCenter
+            
+            dynamicEntityLayer.addLabelDefinition(labelDefinition)
+            dynamicEntityLayer.labelsAreEnabled = true
+            
+            map.addOperationalLayer(dynamicEntityLayer)
+        }
+    }
+}
+
+private extension Array where Element == Field {
+    /// An array of fields that match the attributes of each observation in the data source.
+    ///
+    /// This schema is derived from the first row in the custom data source.
+    static var vesselFields: [Field] = [
+        Field(type: .text, name: "MMSI", alias: "MMSI", length: 256),
+        Field(type: .float64, name: "SOG", alias: "SOG", length: 8),
+        Field(type: .float64, name: "COG", alias: "COG", length: 8),
+        Field(type: .text, name: "VesselName", alias: "VesselName", length: 256),
+        Field(type: .text, name: "CallSign", alias: "CallSign", length: 256),
+        Field(type: .text, name: "globalid", alias: "globalid", length: 256)
+    ]
+}
+
+/// The vessel feed that is emitting custom dynamic entity events.
+private struct VesselFeed: CustomDynamicEntityFeed {
+    let events: AsyncPublisher<AnyPublisher<CustomDynamicEntityFeedEvent, Never>>
+    
+    /// Creates a feed that is receiving the events that are being passed in.
+    ///
+    /// The events will be emitted with a delay so it resembles live data.
+    /// - Parameter events: The events that will be emitted by the feed
+    init<S>(events: S) where S: Sequence, S.Element == CustomDynamicEntityFeedEvent {
+        self.events = events.publisher
+            .delay(for: 1, scheduler: RunLoop.main)
+            .eraseToAnyPublisher()
+            .values
+    }
+}
+
+private extension URL {
+    /// The URL to the selected vessels JSON data.
+    static var selectedVesselsDataSource: URL {
+        Bundle.main.url(
+            forResource: "AIS_MarineCadastre_SelectedVessels_CustomDataSource",
+            withExtension: "json"
```

Comment:
This really is a misnomer as the file is not actually valid JSON syntax. It really is just a text file that contains JSON snippets. But I'm guessing there's nothing that we can do about that?

Reply:
I can coordinate with other platforms to fix the JSON. Is it OK if I simply add a `[]` to the whole file to make it an array?

Reply:
It would also need commas at the end of every line (except for the last one), right? Although, that could inhibit reading and parsing the vessels in line by line, at least without some massaging.

Reply:
I agree, we should keep the current line by line structure. It's so called [JSON Lines](https://jsonlines.org/) ([SO thread](https://stackoverflow.com/questions/12451431/loading-and-parsing-a-json-file-with-multiple-json-objects)) and has a recommended file extension `.jsonl`. Do you want me to change it to that?

Reply:
Interesting, I haven't heard of that before. If that is the better way to represent this file then I am ok with the file extension change.